wagtail/contrib/redirects/base_formats.py

```diff
@@ -0,0 +1,228 @@
+        return dataset.export(self.get_title(), **kwargs)
+
+    def get_extension(self):
+        return self.get_format().extensions[0]
+
+    def get_content_type(self):
+        return self.CONTENT_TYPE
+
+    def can_import(self):
+        return hasattr(self.get_format(), 'import_set')
+
+    def can_export(self):
+        return hasattr(self.get_format(), 'export_set')
+
+
+class TextFormat(TablibFormat):
+    def get_read_mode(self):
+        return 'r'
+
+    def is_binary(self):
+        return False
+
+
+class CSV(TextFormat):
+    TABLIB_MODULE = 'tablib.formats._csv'
+    CONTENT_TYPE = 'text/csv'
+
+    def create_dataset(self, in_stream, **kwargs):
+        return super().create_dataset(in_stream, **kwargs)
+
+
+class JSON(TextFormat):
+    TABLIB_MODULE = 'tablib.formats._json'
+    CONTENT_TYPE = 'application/json'
+
+
+class YAML(TextFormat):
+    TABLIB_MODULE = 'tablib.formats._yaml'
+    # See https://stackoverflow.com/questions/332129/yaml-mime-type
+    CONTENT_TYPE = 'text/yaml'
+
+
+class TSV(TextFormat):
+    TABLIB_MODULE = 'tablib.formats._tsv'
+    CONTENT_TYPE = 'text/tab-separated-values'
+
+    def create_dataset(self, in_stream, **kwargs):
+        return super().create_dataset(in_stream, **kwargs)
+
+
+class ODS(TextFormat):
+    TABLIB_MODULE = 'tablib.formats._ods'
+    CONTENT_TYPE = 'application/vnd.oasis.opendocument.spreadsheet'
+
+
+class HTML(TextFormat):
+    TABLIB_MODULE = 'tablib.formats._html'
+    CONTENT_TYPE = 'text/html'
+
+
+class XLS(TablibFormat):
+    TABLIB_MODULE = 'tablib.formats._xls'
+    CONTENT_TYPE = 'application/vnd.ms-excel'
+
+    def create_dataset(self, in_stream):
+        """
+        Create dataset from first sheet.
+        """
+        import xlrd
+        xls_book = xlrd.open_workbook(file_contents=in_stream)
+        dataset = tablib.Dataset()
+        sheet = xls_book.sheets()[0]
+
+        dataset.headers = sheet.row_values(0)
+        for i in range(1, sheet.nrows):
+            dataset.append(sheet.row_values(i))
+        return dataset
+
+
+class XLSX(TablibFormat):
+    TABLIB_MODULE = 'tablib.formats._xlsx'
+    CONTENT_TYPE = 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet'
+
+    def create_dataset(self, in_stream):
+        """
+        Create dataset from first sheet.
+        """
+        from io import BytesIO
+        import openpyxl
+        xlsx_book = openpyxl.load_workbook(BytesIO(in_stream), read_only=True)
+
+        dataset = tablib.Dataset()
+        sheet = xlsx_book.active
+
+        # obtain generator
+        rows = sheet.rows
+        dataset.headers = [cell.value for cell in next(rows)]
+
+        for row in rows:
+            row_values = [cell.value for cell in row]
+            dataset.append(row_values)
+        return dataset
+
+
+#: These are the default formats for import and export. Whether they can be
+#: used or not is depending on their implementation in the tablib library.
```

Comment:
This comment references export, is this just for future or is there a way to export (I could not see it)

Reply:
@lb- So this module is a special case, since its actually from another Django library and I've tried to not to make any changes to it, mostly to make updating easier. Would you be ok keeping the comment based on my reasoning?

Reply:
Oh yep. I can see that now. 100% agree. 
Also, we might even want to bring this util into core eventually and use for similar shared import/export functionality. For now - this is great, leave as is.